### PR TITLE
Add command to deploy libraries on the origin chain

### LINKED ISSUE #56: Libraries like Message bus, MPP and GatewayLib should be deployed once per chain
As a chain owner, 
When I set up a new auxiliary chain, 
It should be possible to use already deployed libraries on origin chain so that chain setup gas cost can be reduced.

### PR DESCRIPTION
**Pull request by sarvesh-ost**
Fixes #56  

🚫 Merge after #76 

This PR added a command to deploy libraries on the origin chain and also write the deployed libraries addresses in the mosaic config. 

Using deployed libraries while setting up an auxiliary chain from the mosaic-config should be done in the ticket #77

### CODE REVIEW

**Comment by schemar** [comment]
Reviewing 🦑 
Together with #76

File: src/bin/mosaic-libraries.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env node
+
+import * as commander from 'commander';
+
+import { version } from '../../package.json';
```
Comment (schemar) [inline src/bin/mosaic-libraries.ts]: All references to `package.json` have to be removed. See #75 ("version and description References")

File: src/bin/mosaic-libraries.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      } = await OriginChain.deployLibraries(originWeb3, deployer);
+
+      const mosaicConfig:MosaicConfig = new MosaicConfig();
+
+      mosaicConfig.originChain.chain = chain;
+      mosaicConfig.originChain.contractAddress.gatewayLibAddress = gatewayLib.address;
+      mosaicConfig.originChain.contractAddress.messageBusAddress = messageBus.address;
+      mosaicConfig.originChain.contractAddress.merklePatricialLibAddress = merklePatriciaProof.address;
+
+      mosaicConfig.writeToMosaicConfigDirectory();
```
Comment (schemar) [inline src/bin/mosaic-libraries.ts]: It seems you are overwriting any existing mosaic config. Instead, you should append to any existing config.

File: src/bin/mosaic.ts [hunk 3]
```diff
@@ -12,4 +12,5 @@ mosaic
   .command('logs <chain>', 'follow the logs in the container of the given chain')
   .command('list', 'list all running containers that run a mosaic chain')
   .command('create <new-chain-id> <origin-websocket> <password-file>', 'creates a new auxiliary chain on the given origin chain')
+  .command('libraries <chain> <origin-websocket>', 'Deploys libraries on a origin chain')
```
Comment (schemar) [inline src/bin/mosaic.ts]:
```suggestion
  .command('libraries <chain> <origin-websocket> <deployer>', 'Deploys libraries on a origin chain')
```

File: src/bin/mosaic-libraries.ts (hunk 1 above)
Comment (sarvesh-ost) [reply src/bin/mosaic-libraries.ts]: OK, I will make this change if #75 is merged before.  👼

Comment (schemar) [reply src/bin/mosaic-libraries.ts]: You have to remove it either way, the change does not depend on #75.

Comment (sarvesh-ost) [reply src/bin/mosaic-libraries.ts]: Done

File: src/bin/mosaic-libraries.ts (hunk 2 above)
Comment (sarvesh-ost) [reply src/bin/mosaic-libraries.ts]: Done

File: src/bin/mosaic.ts (hunk 3 above)
Comment (sarvesh-ost) [reply src/bin/mosaic.ts]: Done

**Comment by sarvesh-ost** [comment]
Ready for review again.

**Comment by abhayks1** [comment]
On it 🏃 🐇

File: src/Config/MosaicConfigV2.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
+import * as path from 'path';
```
Comment (schemar) [inline src/Config/MosaicConfigV2.ts]: Now that #76 is merged, why does this appear as a whole new file? 🤔 Can you please rebase on latest `master` so we can see what's going on here?

Comment (sarvesh-ost) [reply src/Config/MosaicConfigV2.ts]: It doesn't show me as a new file. I have rearranged the sequence of classes to fix linter issues.

**Pull request merged**